### DISCUSSION TITLE
Problem running pyscenic through sbatch

### DISCUSSION THREAD

**Discussion by GeoKont** [aertslab/SCENIC#444] [Python]
Hello,

Thanks for developing this great tool.
I am trying to run it using sbatch on remote HPC.
I've installed pyscenic v.0.12.1 using pip after IT's suggestion; it seems to run properly on the login CLI.
When I try to run it with sbatch, no steps are running, with no error message. HPC's support informed me that the specific job seems to require internet connection, so nothing is done since the nodes do not have Internet access. 

Does the aucell step require internet access for some reason? And if so, is there a way to run it through sbatch with no internet access?

The first two steps, arboreto and ctx were run on a different server (128GB), but I get an error for aucell due to memory limitations.

The code I use with sbatch is this:
```
#!/bin/bash

#BATCH --job-name=pyscenic_total
#SBATCH --mem=256G
#SBATCH -c 12
#SBATCH --time=96:00:00
#SBATCH --ntasks=1
#SBATCH --nodes=1
#SBATCH --job-name=pyscenic_total
#SBATCH --error=aucell.%j.err

#dir
cd /work2/pyscenic/pyscenic2

#initiate pyscenic
module purge
module load gnu/8 intelmpi/2018 python/3.8.13 hdf5/1.8.22/gnu arrow/13.0.0
source bin/activate

#run aucell
pyscenic aucell data.loom out/CTX_regulons.csv --output out/AUCell_filtered.loom --num_workers 12

```

Though it starts to run, nothing is done and just stops after reaching time limit.

```
[X@login01 pyscenic2]$ sstat -j 1591733
       JobID  MaxVMSize  MaxVMSizeNode  MaxVMSizeTask  AveVMSize     MaxRSS MaxRSSNode MaxRSSTask     AveRSS MaxPages MaxPagesNode   MaxPagesTask   AvePages     MinCPU MinCPUNode MinCPUTask     AveCPU   NTasks AveCPUFreq ReqCPUFreqMin ReqCPUFreqMax ReqCPUFreqGov ConsumedEnergy  MaxDiskRead MaxDiskReadNode MaxDiskReadTask  AveDiskRead MaxDiskWrite MaxDiskWriteNode MaxDiskWriteTask AveDiskWrite 
------------ ---------- -------------- -------------- ---------- ---------- ---------- ---------- ---------- -------- ------------ -------------- ---------- ---------- ---------- ---------- ---------- -------- ---------- ------------- ------------- ------------- -------------- ------------ --------------- --------------- ------------ ------------ ---------------- ---------------- ------------ 
sstat: error: no steps running for job 1591733
```

Any help would be greatly appreciated!

**Comment by ghuls** [comment]
As far as I know there should not be a direct dependency on an internet connection, unless it is hidden in one of the dependencies.
Can HPC support tell you which URL it is trying to open?
